blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/Containers.java

```diff
@@ -36,7 +37,14 @@ public T get(String name) {
 
             @Override
             public Iterator<T> iterator(int start, int limit) {
-                return base.subList(start,start+limit).iterator();
+                if (start >= base.size()) {
```

Comment:
@kzantow This change looks good. I think there might be other places similar logic needs  to be applied, if so use a common mechanism or List wrapper to return subList with check for stat/end?